### PR TITLE
chore: lint `lifecycle` crate

### PR DESCRIPTION
**Pull request by crepererum**

### CODE REVIEW

**Review by tustvold — APPROVED** [review]
Thanks :+1:

File: lifecycle/src/guard.rs [hunk 1]
```diff
@@ -126,6 +129,12 @@ impl<'a, P, D> LifecycleReadGuard<'a, P, D> {
     }
 }
 
+impl<'a, P, D> Debug for LifecycleReadGuard<'a, P, D> {
```
Comment (tustvold) [inline lifecycle/src/guard.rs:132]: It's pretty minor, but if `P` and/or `D` implement Debug this could print them

Comment (crepererum) [reply lifecycle/src/guard.rs:132]: The problem though is that we don't require that in general (yet), and just requiring it for the `Debug` impl here makes the `Debug` impl conditional, which I think isn't too nice as well.

Comment (tustvold) [reply lifecycle/src/guard.rs:132]: FWIW that's what `#derive[Debug]` does. It's particularly annoying for `#derive[Clone]` impls where it requires the type paratemeters be cloneable, even if they're used in something like an `Arc`. - https://github.com/rust-lang/rust/issues/26925

**Pull request merged**